### PR TITLE
nextcloud-client: update to 3.4.2

### PR DESCRIPTION
**Pull request by gbeliako**
#### Testing the changes
- I tested the changes in this PR: **YES**

 
#### Local build testing
- I built this PR locally for my native architecture, (x86_64-libc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
(CROSSBUILD)
  - armv7hf-musl 
  - armv7lhf
  - armv6hf

### CODE REVIEW

File: srcpkgs/nextcloud-client/template [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 revision=1
 wrksrc="desktop-${version}"
 build_style=cmake
 configure_args="-Wno-dev"
-hostmakedepends="pkg-config inkscape"
-makedepends="qt5-tools-devel qt5-declarative-devel qt5-webchannel-devel
- qt5-location-devel qtkeychain-qt5-devel sqlite-devel libcloudproviders-devel
+hostmakedepends="pkg-config inkscape "
+makedepends="qt5-tools-devel qt5-declarative-devel qt5-webchannel-devel openssl-devel
+ qt5-location-devel qtkeychain-qt5-devel sqlite-devel sqlite libcloudproviders-devel qt5-qmake qt5-host-tools qt5-tools
```
Comment (ericonr) [inline srcpkgs/nextcloud-client/template:10]: Please break these into new lines to keep under 80 columns.

I think the qt5 host stuff should usually go in `hostmakedepends`, too.

File: srcpkgs/nextcloud-client/template [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-version=3.3.5
+version=3.4.2
 revision=1
 wrksrc="desktop-${version}"
 build_style=cmake
 configure_args="-Wno-dev"
-hostmakedepends="pkg-config inkscape"
-makedepends="qt5-tools-devel qt5-declarative-devel qt5-webchannel-devel
- qt5-location-devel qtkeychain-qt5-devel sqlite-devel libcloudproviders-devel
+hostmakedepends="pkg-config inkscape "
```
Comment (ericonr) [inline srcpkgs/nextcloud-client/template:8]: Note you have an extra space here

File: srcpkgs/nextcloud-client/template [hunk 3]
```diff
@@ -39,6 +39,14 @@ if [ "$CROSS_BUILD" ]; then
 	# provides desktoptojson
 	hostmakedepends+=" $(vopt_if dolphin 'kcoreaddons')"
 fi
+if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	#provides argp library
```
Comment (ericonr) [inline srcpkgs/nextcloud-client/template:43]: This comment isn't necessary.

File: srcpkgs/nextcloud-client/template [hunk 4]
```diff
@@ -39,6 +39,14 @@ if [ "$CROSS_BUILD" ]; then
...
+	makedepends+=" argp-standalone"
+fi
+if [ "$XBPS_TARGET_MACHINE" != "armv7l" ]; then
+	#these packages are broken on that architecture
+	hostmakedepends+=" qt5-devel qt5-webengine-devel qt5-webview-devel"
+fi
```
Comment (ericonr) [inline srcpkgs/nextcloud-client/template:49]: Did you see this build break?

Comment (gbeliako) [reply srcpkgs/nextcloud-client/template]: before when building it said that it was unable to build the package as it was broken for some architecture, but it appears to be working now and i was unable to reproduce it so i removed those lines

**Pull request closed without merging**